packages/ui/src/theme/tokens/components/fieldControl.ts

```diff
@@ -1,28 +1,101 @@
+  paddingBlockStart: DesignToken<SpaceValue>;
+  paddingBlockEnd: DesignToken<SpaceValue>;
+  paddingInlineStart: DesignToken<SpaceValue>;
+  paddingInlineEnd: DesignToken<SpaceValue>;
+}
+
+interface FieldControlFocusTokens {
+  borderColor: DesignToken<ColorValue>;
+  boxShadow: DesignToken<BoxShadowValue>;
+}
+
+interface FieldControlDisabledTokens {
+  color: DesignToken<ColorValue>;
+  cursor: DesignToken<CursorValue>;
+  borderColor: DesignToken<ColorValue>;
+  backgroundColor: DesignToken<ColorValue>;
+}
+
+interface FieldControlErrorTokens {
+  borderColor: DesignToken<ColorValue>;
+  _focus: FieldControlErrorFocusTokens;
+}
+
+interface FieldControlErrorFocusTokens {
+  boxShadow: DesignToken<BoxShadowValue>;
+}
+
+interface FieldControlQuietTokens {
+  borderStyle: DesignToken<BorderStyleValue>;
+  borderBlockEnd: DesignToken<BorderValue>;
+  borderInlineStart: DesignToken<BorderValue>;
+  borderInlineEnd: DesignToken<BorderValue>;
+  borderBlockStart: DesignToken<BorderValue>;
+  borderRadius: DesignToken<BorderRadiusValue>;
+  _focus: FieldControlQuietFocusTokens;
+  _error: FieldControlQuietErrorTokens;
+}
+
+interface FieldControlQuietFocusTokens {
+  borderBlockEndColor: DesignToken<ColorValue>;
+  boxShadow: DesignToken<BoxShadowValue>;
+}
+
+interface FieldControlQuietErrorTokens {
+  borderBlockEndColor: DesignToken<ColorValue>;
+  _focus: FieldControlQuietErrorFocusTokens;
+}
+
+interface FieldControlQuietErrorFocusTokens {
+  boxShadow: DesignToken<BoxShadowValue>;
+}
+
 export interface FieldControlTokens {
-  borderStyle: any;
-  borderColor: any;
-  borderWidth: any;
-  borderRadius: any;
-  color: any;
-  paddingBlockStart: any;
-  paddingBlockEnd: any;
-  paddingInlineStart: any;
-  paddingInlineEnd: any;
-
-  fontSize: any;
-  lineHeight: any;
-  transitionDuration: any;
-  outlineColor: any;
-  outlineStyle: any;
-  outlineWidth: any;
-  outlineOffset: any;
-
-  small: any;
-  large: any;
-  quiet: any;
-  _focus: any;
-  _disabled: any;
-  _error: any;
+  borderStyle: DesignToken<BorderStyleValue>;
+  borderColor: DesignToken<ColorValue>;
+  borderWidth: DesignToken<BorderWidthValue>;
+  borderRadius: DesignToken<BorderRadiusValue>;
+  color: DesignToken<ColorValue>;
+  paddingBlockStart: DesignToken<SpaceValue>;
+  paddingBlockEnd: DesignToken<SpaceValue>;
+  paddingInlineStart: DesignToken<SpaceValue>;
+  paddingInlineEnd: DesignToken<SpaceValue>;
+
+  fontSize: DesignToken<FontSizeValue>;
+  lineHeight: DesignToken<LineHeightValue>;
+  transitionDuration: DesignToken<TransitionDurationValue>;
+  outlineColor: DesignToken<OutlineColorValue>;
+  outlineStyle: DesignToken<OutlineStyleValue>;
+  outlineWidth: DesignToken<OutlineWidthValue>;
+  outlineOffset: DesignToken<OutlineOffsetValue>;
+
+  small: FieldControlSizeTokens;
+  large: FieldControlSizeTokens;
+  quiet: FieldControlQuietTokens;
+  _focus: FieldControlFocusTokens;
+  _disabled: FieldControlDisabledTokens;
+  _error: FieldControlErrorTokens;
```

Comment:
Thinking out loud here... would it be more readable to inline these interfaces that aren't shared? `FieldControlErrorTokens` is only used here as far as I can tell. Something we might want to do across the board (if we think it is a good idea). I'm not super opinionated one way or the other, just wanted to bring it up

Reply:
Honestly went back and forth on this a few times. 😅 I decided to create an interface for every one because it was too hard to decide _which_ ones to split out and which ones not to split out. By creating an interface for all of them, it's more consistent.